button_handler.py

```diff
@@ -170,51 +280,61 @@ def update(self) -> list[str]:
 
         :rtype: list[str]
         """
-        current_time = time.monotonic()
-        self._is_pressed = not self._button.value
-
-        # Debounce logic
-        if self._is_pressed != self._was_pressed:
-            time.sleep(self.debounce_time)
-            self._is_pressed = not self._button.value
-
-        events = []
-
-        if self._is_pressed and not self._was_pressed:  # Button just pressed
-            self._press_start_time = current_time
-            if self._first_press_time is None:
-                self._first_press_time = current_time
-            self._press_count += 1
-
-        if (
-            self._is_pressed and current_time - self._press_start_time >= self.long_press_threshold
-        ):  # Holding
-            self._is_holding = True
-            events.append("HOLDING")
-
-        if not self._is_pressed and self._was_pressed:  # Button just released
-            if current_time - self._press_start_time < self.long_press_threshold:
-                if not self.enable_double_press:
-                    events.append("SHORT_PRESS")
-                elif self._press_count == 2:
-                    events.append("DOUBLE_PRESS")
-                else:
-                    self._was_pressed = self._is_pressed
-                    return events
+        inputs = set()
+
+        inputs.update(self._handle_buttons())
+
+        event = self._event
+        event_queue = self._event_queue
+        i = 0
+        while i < len(event_queue):
+            event_queue.get_into(event)
+            input_ = self._handle_event(event)
+            if input_:
+                inputs.add(input_)
+
+        return inputs
+
+    def _handle_buttons(self) -> set(ButtonInput):
```

Comment:
This is not a valid type hint
```suggestion
    def _handle_buttons(self) -> set[ButtonInput]:
```